internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -283,7 +278,88 @@ func createRun(opts *CreateOpts) error {
 			if err := opts.Output.Display(newRotatingSecretsDisplayer(true).PreviewRotatingSecrets(resp.Payload.Config)); err != nil {
 				return err
 			}
+
+		default:
+			return fmt.Errorf("unsupported rotating secret provider type")
+		}
+
+	case secretTypeDynamic:
+		sc, err := readConfigFile(opts)
+		if err != nil {
+			return fmt.Errorf("failed to process config file: %w", err)
+		}
+
+		missingFields := validateSecretConfig(sc)
+
+		if len(missingFields) > 0 {
+			return fmt.Errorf("missing required field(s) in the config file: %s", missingFields)
+		}
+
+		switch sc.Type {
+		case integrations.AWS:
+
+			missingDetails := validateDetails(sc.Details, AwsRequiredKeys)
+
+			if len(missingDetails) > 0 {
+				return fmt.Errorf("missing required field(s) in the config file details: %s", missingDetails)
+			}
+
+			var role AwsAssumeRole
```

Comment:
Would be good to follow the same pattern for GCP but with the nested `service_account_impersonation` fields similar to AWS `assume_role so that it's consistent.

Reply:
Great point!